lib/datasampler.rb

```diff
@@ -25,70 +41,116 @@ def delete_data
   def import_clients
-    # get clients from file, import to Client model
     clients_filename = @directory + 'clients.json'
-    clients_json = File.open(clients_filename, &:readline)
-    clients = Client.array_from_json(clients_json,
-                                     fields: Client.columns.map(&:to_s))
-
-    clients.each(&:save)
+    Client.unrestrict_primary_key
+    iterate_lines(clients_filename) do |line|
+      Client.create(JSON.parse(line))
+    end
+    Client.restrict_primary_key
   end
 
   # Imports the projects from projects.json
   def import_projects
-    # get projects
     projects_filename = @directory + 'projects.json'
-    projects_json = File.open(projects_filename, &:readline)
-    projects = Project.array_from_json(projects_json,
-                                       fields: Project.columns.map(&:to_s))
+    Project.unrestrict_primary_key
+    iterate_lines(projects_filename) do |line|
+      Project.create(JSON.parse(line))
+    end
+    Project.restrict_primary_key
+  end
 
-    projects.each(&:save)
+  # Loop through measurements and import their datat
+  def import_measurements(measurements)
+    measurements.each do |measurement_name|
+      table = Plugin.where(name: measurement_name).first.storage_table
+      filename = @directory + table + '.json'
+      puts "  Importing measurement #{measurement_name}"
+      iterate_lines(filename) do |line|
+        JSON.parse(line).each do |key, value|
+          date = update_date(value, @date_offset)
+          measurement[key] = date unless date.nil?
+        end
+        Iam.settings.DB[table.to_sym].insert(measurement)
+      end
+    end
   end
 
   # Loop through our defined resources and import the data for each one
-  # rubocop:disable MethodLength
   def import_resources
     # for each resource type, look for a file  of measurement data
     Report.plugin_matrix.each do |resource_name, measurements|
       filename = @directory + resource_name + '.json'
-      resource_json = File.open(filename, &:readline)
-      model = Object.const_get((resource_name + 'Resource').camelcase(:upper))
-      resources = model.array_from_json(resource_json,
-                                        fields: model.columns.map(&:to_s))
 
-      resources.each(&:save)
-      measurements.each do |measurement_name|
-        table = Plugin.where(name: measurement_name).first.storage_table
-        filename = @directory + table + '.json'
-        measurement_data = JSON.parse(File.open(filename, &:readline))
-        Iam.settings.DB[table.to_sym].multi_insert(measurement_data)
+      next unless File.exist? filename
+      puts "Importing resource #{resource_name} from #{filename}"
+
+      model = Object.const_get((resource_name + 'Resource').camelcase(:upper))
+      model.unrestrict_primary_key
+      iterate_lines(filename) do |line|
+        resource = JSON.parse(line)
+        resource.each do |key, value|
+          value = update_date(value, @date_offset)
```

Comment:
We probably don't need to update the date on resources - it shouldn't hurt anything, but we only ever care about the measurement time